tests/backup/backup_portworx_test.go

```diff
@@ -582,3 +584,279 @@ var _ = Describe("{RestoreEncryptedAndNonEncryptedBackups}", func() {
+					periodicSchedulePolicyNames = append(periodicSchedulePolicyNames, periodicSchedulePolicyName)
+					periodicSchedulePolicyUid = uuid.New()
+					periodicSchedulePolicyUids = append(periodicSchedulePolicyUids, periodicSchedulePolicyUid)
+					periodicSchedulePolicyInfo := Inst().Backup.CreateIntervalSchedulePolicy(5, 15, 5)
+					err = Inst().Backup.BackupSchedulePolicy(periodicSchedulePolicyName, periodicSchedulePolicyUid, orgID, periodicSchedulePolicyInfo)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of periodic schedule policy of interval 15 minutes named [%s]", periodicSchedulePolicyName))
+					periodicSchedulePolicyUid, err = Inst().Backup.GetSchedulePolicyUid(orgID, ctx, periodicSchedulePolicyName)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching uid of periodic schedule policy named [%s]", periodicSchedulePolicyName))
+				})
+				Step("Resize the volume before backup schedule", func() {
+					log.InfoD("Resize the volume before backup schedule")
+					for _, ctx := range contexts {
+						var appVolumes []*volume.Volume
+						log.InfoD(fmt.Sprintf("get volumes for %s app", ctx.App.Key))
+						appVolumes, err := Inst().S.GetVolumes(ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Fetch volumes for app %s", ctx.App.Key))
+						dash.VerifyFatal(len(appVolumes) > 0, true, "Verifying if app volumes exist")
+						var requestedVols []*volume.Volume
+						log.InfoD(fmt.Sprintf("Increase volume size %s on app %s",
+							Inst().V.String(), ctx.App.Key))
+						requestedVols, err = Inst().S.ResizeVolume(ctx, Inst().ConfigMap)
+						dash.VerifyFatal(err, nil, "Verifying volume resize")
+						log.InfoD(fmt.Sprintf("validate successful volume size increase on app %s",
+							ctx.App.Key))
+						for _, volume := range requestedVols {
+							// Need to pass token before validating volume
+							params := make(map[string]string)
+							if Inst().ConfigMap != "" {
+								params["auth-token"], err = Inst().S.GetTokenFromConfigMap(Inst().ConfigMap)
+								dash.VerifyFatal(err, nil, "Fetching token from configMap")
+							}
+							err := Inst().V.ValidateUpdateVolume(volume, params)
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Validate volume %v update status", volume))
+						}
+					}
+				})
+				Step("Create schedule backup immediately after initiating volume resize", func() {
+					log.InfoD("Create schedule backup after initiating volume resize")
+					ctx, err := backup.GetAdminCtxFromSecret()
+					dash.VerifyFatal(err, nil, "Fetching px-central-admin ctx")
+					scheduleName = fmt.Sprintf("%s-schedule-%v", BackupNamePrefix, time.Now().Unix())
+					scheduleNames = append(scheduleNames, scheduleName)
+					preRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, preRuleNameList[0])
+					postRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, postRuleNameList[0])
+					err = CreateScheduleBackup(scheduleName, SourceClusterName, backupLocationName, backupLocationUID, []string{namespace},
```

Comment:
This comment might be beyond the scope of this PR, If the appList parameter contains two applications, the rules of the first app will be applied to the second app as well. Using preRuleNameList[index] and postRuleNameList[index] will only work when Inst().GlobalScaleFactor is set to 1. It would be more appropriate for preRuleNameList and postRuleNameList to be represented as a map that links the appName to its respective preRule and postRule.

Reply:
Good observation.
It should not matter for this PR , since it is mainly intended or focusing on volume resize